src/config.rs

```diff
@@ -526,61 +526,30 @@ impl<'a> Cfg<'a> {
         &self,
         force_install_active: Option<bool>,
     ) -> Result<Option<(LocalToolchainName, ActiveReason)>> {
-        let (components, targets, profile, toolchain, reason) = match self.find_override_config()? {
-            Some((
-                OverrideCfg::Official {
-                    components,
-                    targets,
-                    profile,
-                    toolchain,
-                },
-                reason,
-            )) => (components, targets, profile, toolchain, reason),
-            Some((override_config, reason)) => {
-                return Ok(Some((override_config.into_local_toolchain_name(), reason)));
-            }
-            None => {
-                return Ok(self
-                    .get_default()?
-                    .map(|x| (x.into(), ActiveReason::Default)));
-            }
-        };
-
         let should_install_active = if let Some(force) = force_install_active {
             force
         } else {
             self.should_auto_install()?
         };
 
-        if !should_install_active {
-            return Ok(Some(((&toolchain).into(), reason)));
+        if should_install_active {
+            return self
+                .find_or_install_active_toolchain(false, false)
+                .await
+                .map(Some);
         }
+        self.active_toolchain()
+    }
 
-        let components = components.iter().map(AsRef::as_ref).collect::<Vec<_>>();
-        let targets = targets.iter().map(AsRef::as_ref).collect::<Vec<_>>();
-        match DistributableToolchain::new(self, toolchain.clone()) {
-            Err(RustupError::ToolchainNotInstalled { .. }) => {
-                DistributableToolchain::install(
-                    self,
-                    &toolchain,
-                    &components,
-                    &targets,
-                    profile.unwrap_or_default(),
-                    false,
-                )
-                .await?;
-            }
-            Ok(mut distributable) => {
-                if !distributable.components_exist(&components, &targets)? {
-                    distributable
-                        .update(&components, &targets, profile.unwrap_or_default())
-                        .await?;
-                }
-            }
-            Err(e) => return Err(e.into()),
-        };
-
-        Ok(Some(((&toolchain).into(), reason)))
+    pub(crate) fn active_toolchain(&self) -> Result<Option<(LocalToolchainName, ActiveReason)>> {
```

Comment:
We have to split out a sync function because v1.28.1 has made the above async again, but we really need to find the active toolchain in a sync context in the commits that follow... (sigh)

Reply:
Also, I suggest renaming `Cfg::find_active_toolchain()` to something else.

Reply:
To be honest I'm struggling to think of a good name.

Reply:
@ChrisDenton I'm currently thinking about something like:

| Before | After | Remarks |
|---|---|---|
| find_active_toolchain() | maybe_install_active_toolchain() | Dispatching function |
| N/A | active_toolchain() | Queries the active toolchain |
| find_or_install_active_toolchain() | install_active_toolchain() | Installs the active toolchain if uninstalled |

Reply:
That sounds good to me. I think those names do make the relationship between them clearer to me at least.